vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -106,19 +88,53 @@ class Dropdown(VizroBaseModel):
     _validate_options = model_validator(mode="before")(validate_options_dict)
 
     def __call__(self, options):
-        full_options, default_value = get_options_and_default(options=options, multi=self.multi)
-        option_height = _calculate_option_height(full_options)
-        altered_options = _add_select_all_option(full_options=full_options) if self.multi else full_options
+        dict_options, default_value = get_dict_options_and_default(options=options, multi=self.multi)
+        option_height = _calculate_option_height(dict_options)
+
+        value = self.value if self.value is not None else default_value
+
+        if self.multi:
+            value = value if isinstance(value, list) else [value]  # type: ignore[assignment]
+            dict_options = [
+                {
+                    "label": dbc.Checkbox(
+                        id=f"{self.id}_select_all",
+                        value=len(value) == len(dict_options),  # type: ignore[arg-type]
+                        label="Select All",
+                        persistence=True,
+                        persistence_type="session",
+                        className="dropdown-select-all",
+                    ),
+                    # Special sentinel value used in update_dropdown_select_all.
+                    # This never gets sent to the server.
+                    "value": "__SELECT_ALL",
+                },
+                *dict_options,
+            ]
+
+            clientside_callback(
+                ClientsideFunction(namespace="dropdown", function_name="update_dropdown_select_all"),
+                output=[
+                    Output(f"{self.id}_select_all", "value"),
+                    Output(self.id, "value"),
+                ],
+                inputs=[
+                    Input(self.id, "value"),
+                    State(self.id, "options"),
+                ],
+                prevent_initial_call=True,
+            )
 
         return html.Div(
             children=[
                 dbc.Label(self.title, html_for=self.id) if self.title else None,
                 dcc.Dropdown(
                     id=self.id,
-                    options=altered_options,
-                    value=self.value if self.value is not None else default_value,
+                    options=dict_options,
+                    value=value,
                     multi=self.multi,
                     optionHeight=option_height,
+                    clearable=False,
                     persistence=True,
                     persistence_type="session",
```

Comment:
Minor visual improvement, such that there is no blank space when nothing is selected. This is also according to our QB designs:

```suggestion
                    persistence_type="session",
                    placeholder="Select option",
```